app/views/users/account.html.erb

```diff
@@ -10,166 +10,104 @@
   </ul>
 <% end %>
 
-<%= error_messages_for current_user %>
-<%= form_for current_user, :url => { :action => :account }, :method => :post, :html => { :multipart => true, :id => "accountForm", :class => "standard-form", :autocomplete => :off } do |f| %>
-  <fieldset>
-    <div class="standard-form-row">
-      <label class="standard-label"><%= t "users.new.display name" %></label>
-      <%= f.text_field :display_name %>
-    </div>
-  </fieldset>
-
-  <fieldset>
-    <div class="standard-form-row">
-      <label class="standard-label"><%= t ".current email address" %></label>
-      <input type="email" disabled value="<%= current_user.email %>" />
-      <span class="form-help deemphasize"><%= t ".email never displayed publicly" %></span>
-    </div>
-
-    <div class="standard-form-row">
-      <label class="standard-label"><%= t ".new email address" %></label>
-      <%= f.email_field :new_email, :autocomplete => :off %>
-      <span class="form-help deemphasize"><%= t ".email never displayed publicly" %></span>
-    </div>
-  </fieldset>
-
-  <fieldset>
-    <div class="standard-form-row">
-        <label class="standard-label"><%= t "users.new.password" %></label>
-      <%= f.password_field :pass_crypt, :value => "", :autocomplete => :off %>
-    </div>
-
-    <div class="standard-form-row">
-      <label class="standard-label"><%= t "users.new.confirm password" %></label>
-      <%= f.password_field :pass_crypt_confirmation, :value => "", :autocomplete => :off %>
+<%= bootstrap_form_for current_user, :url => { :action => :account }, :method => :post, :html => { :multipart => true, :id => "accountForm", :autocomplete => :off } do |f| %>
+
+  <%= f.text_field :display_name %>
+  <%= f.email_field :email, :disabled => true, :label => t(".current email address") %>
+  <%= f.email_field :new_email, :autocomplete => :off %>
+  <%= f.password_field :pass_crypt, :value => "", :autocomplete => :off %>
+  <%= f.password_field :pass_crypt_confirmation, :value => "", :autocomplete => :off %>
+
+  <fieldset class="form-group">
+    <label for="user_auth_provider"><%= t(".external auth") %></label>
+    <div class="form-row">
+      <%= f.select(:auth_provider, Auth.providers, :hide_label => "true", :wrapper => { :class => "col-auto mb-0" }) %>
+      <%= f.text_field(:auth_uid, :hide_label => true, :wrapper => { :class => "col mb-0" }) %>
```

Comment:
Optionally: to make it more accessible you could add the sr-only class to the label and a descriptive label text.

Reply:
I've added better descriptive labels in 3cabed9 (although auth_uid is still quite a technical concept, and I'm not even sure it's clear what the box is for in non-sr situations, but that's not a focus of this refactoring).

The `:hide_label` feature of bootstrap_form does exactly what you suggest - the label is still in the html tree, just marked with the sr-only class, but otherwise there and linked to the relevant input field :+1:

Reply:
Ah, thanks, did not know this. Good feature.